### PR TITLE
Add decidable type class

### LINKED ISSUE #453: Add Decidable type class
## Description

Add `Decidable` type class.

Links in Arrow:
- [Decidable](https://github.com/arrow-kt/arrow/blob/master/modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Decidable.kt)

## Modules

- Bow (core)

### PR DESCRIPTION
**Pull request by devesh-shetty**
## Related issues
Resolves #453 

## Goal
This PR adds the Decidable type class.

### CODE REVIEW

File: Sources/Bow/Typeclasses/Divisible.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/// Divisible extends Divide by providing an empty value.
+public protocol Divisible: Divide {
```
Comment (devesh-shetty) [inline Sources/Bow/Typeclasses/Divisible.swift]: @truizlop I have branched this off `452-add-divisible-type-class` as I needed `Divisible` to implement `Decidable`

Once that PR is in, I can rebase and Divisible and related files will go away.

File: Sources/Bow/Typeclasses/Decidable.swift [hunk 2]
```diff
@@ -0,0 +1,247 @@
+/// Decidable is a typeclass modeling contravariant decision. Decidable is the contravariant version of Alternative.
+///
+/// Decidable basically states: Given a Kind<F, A> and a Kind<F, B> and a way to turn Z into either A or B it gives you a Kind<F, Z>.
+public protocol Decidable: Divisible {
```
Comment (devesh-shetty) [inline Sources/Bow/Typeclasses/Decidable.swift]: This file and pbxproj are the only changes in this PR, rest are from `452-add-divisible-type-class`

File: Sources/Bow/Typeclasses/Decidable.swift [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            fh,
+            choose(fi, fj, id),
+            f
+        )
+    }
+}
+
+// MARK: - Syntax for Decidable
+public extension Kind where F: Decidable {
+    func choose<B, Z>(_ fb: Kind<F, B>, _ f: (Z) -> Either<A, B>) -> Kind<F, Z> {
```
Comment (truizlop) [inline Sources/Bow/Typeclasses/Decidable.swift]: These functions should be static as we have done in [Applicative](https://github.com/bow-swift/bow/blob/master/Sources/Bow/Typeclasses/Applicative.swift#L583) and [Concurrent](https://github.com/bow-swift/bow/blob/master/Sources/BowEffects/Typeclasses/Concurrent.swift#L40) to have an easier way of using them. Also, the return type should be `Kind<F, A>` (as done in the 2 type classes stated above).

**Comment by truizlop** [comment]
Thanks!

**Pull request merged**